src/main/kotlin/edu/kit/compiler/parser/AbstractParser.kt

```diff
@@ -124,8 +115,15 @@ abstract class AbstractParser(val sourceFile: AnnotatableFile, tokens: Sequence<
      *
      * @param anchorSet [AnchorSet] containing all tokens that are accepted as the next token in the stream.
      */
-    protected fun panicMode(anchorSet: AnchorUnion) {
+    protected fun recover(anchorSet: AnchorUnion) {
         val anc = anchorSet.provide()
         while (peek() !in anc) next()
     }
+
+    /**
+     * Annotate the source input with an error message
+     */
+    protected fun reportError(message: String) {
+        sourceFile.annotate(AnnotationType.ERROR, sourceFile.currentPosition, message)
```

Comment:
Auch wenn `sourceFile.currentPosition` hier wahrscheinlich meistens funktioniert, würde ich trotzdem die position ausm token nehmen. Das Problem ist, dass die Position ausm SourceFile nur dann stimmt, wenn du mit `next()` genau auf dem fehlerhaften Token stehst. Wenn man irgendwo mal `peek()` verwendet und dann ohne `next()` das hier aufruft, ist die Position falsch.

Reply:
fixed in 7c58c90af0966875e9215c7eb200223bec5b7ba1